File: packages/ui-flutter/test/blocks/alert_stack_block_test.dart
import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';
import 'package:kinetix_ui/kinetix_ui.dart';

// The "Alert stack" block — see testimonial_block_test.dart for how block fixtures work.
//
// kx-block:start
class AlertStackBlock extends StatelessWidget {
  const AlertStackBlock({super.key});

  @override
  Widget build(BuildContext context) {
    return const Column(
      mainAxisSize: MainAxisSize.min,
      children: <Widget>[
        KinetixAlert(
          children: <Widget>[
            KinetixAlertTitle('Heads up'),
            KinetixAlertDescription('You can add components to your app using the CLI.'),
          ],
        ),
        SizedBox(height: 12),
        KinetixAlert(
          variant: KinetixAlertVariant.destructive,
          children: <Widget>[
            KinetixAlertTitle('Payment failed'),
            KinetixAlertDescription('Update your billing details to keep your subscription active.'),
          ],
        ),
        SizedBox(height: 12),
        KinetixAlert(
          variant: KinetixAlertVariant.success,
          children: <Widget>[KinetixAlertTitle('Changes saved')],
        ),
      ],
    );
  }
}
// kx-block:end

void main() {
  testWidgets('renders every variant with its own copy', (WidgetTester tester) async {
    await tester.pumpWidget(
      MaterialApp(
        home: KinetixTheme(
          brightness: Brightness.light,
          child: const Scaffold(body: SingleChildScrollView(child: AlertStackBlock())),
        ),
      ),
    );

    expect(find.text('Heads up'), findsOneWidget);
    expect(find.text('Payment failed'), findsOneWidget);
    expect(find.text('Changes saved'), findsOneWidget);
  });
}

File: packages/ui-compose/ui/src/test/kotlin/com/kinetixui/ui/blocks/AlertStackBlockTest.kt
package com.kinetixui.ui.blocks

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.runtime.Composable
import androidx.compose.ui.test.junit4.v2.createComposeRule
import androidx.compose.ui.test.onNodeWithText
import androidx.compose.ui.unit.dp
import com.kinetixui.ui.KinetixAlert
import com.kinetixui.ui.KinetixAlertDescription
import com.kinetixui.ui.KinetixAlertTitle
import com.kinetixui.ui.KinetixAlertVariant
import com.kinetixui.ui.KinetixTheme
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner
import org.robolectric.annotation.Config

// The "Alert stack" block — see TestimonialBlockTest.kt for how block fixtures work.
//
// kx-block:start
@Composable
fun AlertStackBlock() {
    Column(verticalArrangement = Arrangement.spacedBy(12.dp)) {
        KinetixAlert {
            KinetixAlertTitle("Heads up")
            KinetixAlertDescription("You can add components to your app using the CLI.")
        }
        KinetixAlert(variant = KinetixAlertVariant.Destructive) {
            KinetixAlertTitle("Payment failed")
            KinetixAlertDescription("Update your billing details to keep your subscription active.")
        }
        KinetixAlert(variant = KinetixAlertVariant.Success) {
            KinetixAlertTitle("Changes saved")
        }
    }
}
// kx-block:end

@RunWith(RobolectricTestRunner::class)
@Config(sdk = [34])
class AlertStackBlockTest {
    @get:Rule
    val rule = createComposeRule()

    @Test
    fun renders_every_variant_with_its_own_copy() {
        rule.setContent { KinetixTheme(darkTheme = false) { AlertStackBlock() } }
        rule.onNodeWithText("Heads up").assertExists()
        rule.onNodeWithText("Payment failed").assertExists()
        rule.onNodeWithText("Changes saved").assertExists()
    }
}
